test/exe/main_common_test.cc

```diff
@@ -7,65 +7,83 @@
 // TODO(issues/2649): This test needs to be parameterized on IP versions.
 #ifndef ENVOY_CONFIG_COVERAGE
 
+#include "common/runtime/runtime_impl.h"
+
 #include "exe/main_common.h"
 
 #include "server/options_impl.h"
 
 #include "test/test_common/environment.h"
+#include "test/test_common/utility.h"
 
 #include "gtest/gtest.h"
 
 #ifdef ENVOY_HANDLE_SIGNALS
 #include "exe/signal_action.h"
 #endif
 
-#ifdef ENVOY_HOT_RESTART
-#include "server/hot_restart_impl.h"
-#endif
-
 namespace Envoy {
 
-TEST(MainCommon, ConstructDestruct) {
+class MainCommonTest : public testing::Test {
+public:
+  MainCommonTest()
+      : config_file_(Envoy::TestEnvironment::getCheckedEnvVar("TEST_RUNDIR") +
+                     "/test/config/integration/google_com_proxy_port_0.v2.yaml"),
+        random_string_(fmt::format("{}", random_generator_.random() % 1024)),
```

Comment:
It's probably worth pointing out here that (a) I refactored the test so it has a test-class, to share the argv prep code, which became more complex when I added randomized base-IDs.

I needed to add randomized base-IDs to get tests to pass consistently under CI; the theory being that CI was instantiating more than one test at a time on a machine, and things were conflicting.  I wasn't sure what the range of 'base-id' was so I made it 0-1023.  Comments welcome.

Reply:
I think this comment would be useful in the file itself.

Reply:
Indeed, good call.  I added a class comment and also factored out the base-id string generation. Also after chatting with @htuch about why this was needed, I decided getpid() was the critical component of the ID, so I combined it in.  Looking at the base-id impl, we have a full uint32_t to play with, but I decided to give more of the bits to the pid() in my choice of prime.